coordinator/pipeline.go

```diff
@@ -97,7 +97,9 @@ func NewPipeline(ctx context.Context,
 	proversPool := NewProversPool(len(provers))
 	proversPoolSize := 0
 	for _, prover := range provers {
-		if err := prover.WaitReady(ctx); err != nil {
+		ctxTimeout, cancel := context.WithTimeout(ctx, 20*time.Second)
```

Comment:
Maybe leave the interval of the timeout in a separated variable and use it in the context creation. Also, I'd leave a TODO comment on top of it to further analysis to check whether the size of the interval must reside or not in config file.

Reply:
Uploaded a new version